src/main/webapp/app/shared/services/auth.service.ts

```diff
@@ -10,20 +10,36 @@ export async function getAuthToken(firebaseToken: string): Promise<void> {
   const headers = {
     [FIREBASE_AUTH_HEADER_NAME]: firebaseToken
   };
-  const response = await axios.post('/api/firebase/authenticate', null, {
+  const response = await axios.post('/api/authenticate/firebase', null, {
     headers
   });
-  const jwtToken: string = response.data['id_token'];
+  const jwtToken: string = response.data['accessToken'];
+  const refreshToken: string = response.data['refreshToken'];
   Storage.local.set(AUTH_TOKEN_KEY, jwtToken);
-  store.dispatch({ type: ACTION_TYPES.LOGIN });
+  Storage.local.set(REFRESH_TOKEN_KEY, refreshToken);
+  await fetchAccount();
 }
 
 export function logout(): void {
   Storage.local.remove(FIREBASE_TOKEN_KEY);
   Storage.local.remove(AUTH_TOKEN_KEY);
+  Storage.local.remove(REFRESH_TOKEN_KEY);
   store.dispatch({ type: ACTION_TYPES.LOGOUT });
 }
 
+export async function handleUnauthenticated(): Promise<void> {
+  const token = Storage.local.get(AUTH_TOKEN_KEY) || Storage.session.get(AUTH_TOKEN_KEY);
+  const refreshToken = Storage.local.get(REFRESH_TOKEN_KEY) || Storage.session.get(REFRESH_TOKEN_KEY);
+  if (token && refreshToken) {
+    const response = await axios.post(`/api/authenticate/refresh?refreshToken=${refreshToken}`);
```

Comment:
if this throws error, should clear the local storage (AUTH_TOKEN_KEY, REFRESH_TOKEN_KEY), so that line 33 return false and no need to run the whole if block ady

Reply:
resolve in 925de3b100431105e346fe573c227ebf409e3927